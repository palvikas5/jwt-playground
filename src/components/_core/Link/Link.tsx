import React from 'react';
import { StyledLink } from './link.style';

export interface LinkProps {
  href: string;
  children: string | number;
  asButton?: boolean;
}

const Link = React.forwardRef<HTMLAnchorElement, LinkProps>(
  ({ href, asButton, children }: LinkProps, ref) => {
    return (
      <StyledLink asButton={asButton} ref={ref} href={href}>
        {children}
      </StyledLink>
    );
  },
);

Link.defaultProps = {
  asButton: false,
};

export default Link;
