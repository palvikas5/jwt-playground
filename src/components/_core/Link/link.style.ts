/* stylelint-disable value-keyword-case */

import styled from 'styled-components';
import { LinkProps } from './Link';

const StyledLink = styled.a<LinkProps>`
  display: block;
  color: white;
  text-decoration: none;
  border-bottom: 1px solid white;
  ${({ asButton }) =>
    asButton &&
    `
    background-color: #3e8efa;
    text-transform: uppercase;
    padding: 12px 30px;
    border: none;
    border-radius: 3px;
  `}
`;

export { StyledLink };
