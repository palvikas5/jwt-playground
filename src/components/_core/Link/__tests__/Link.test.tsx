import React from 'react';
import { render } from '@testing-library/react';
import Link from '../Link';

describe('<Link />', () => {
  it('should render successfully', () => {
    expect.assertions(1);

    const { container } = render(
      <Link href='https://dummy-link-url'>link-text</Link>,
    );
    expect(container).toMatchSnapshot();
  });

  it('should contains allowed values as children', () => {
    expect.assertions(2);

    const wrapper = render(
      <Link href='https://dummy-link-url'>link-text</Link>,
    );
    expect(wrapper.getByText(/^link-text$/)).toBeInTheDocument();

    const wrapper1 = render(<Link href='https://dummy-link-url'>{1}</Link>);
    expect(wrapper1.getByText(/^1$/)).toBeInTheDocument();
  });

  it('should render successfully as a button', () => {
    expect.assertions(1);

    const { container } = render(
      <Link href='https://dummy-link-url' asButton>
        link-text
      </Link>,
    );
    expect(container).toMatchSnapshot();
  });
});
