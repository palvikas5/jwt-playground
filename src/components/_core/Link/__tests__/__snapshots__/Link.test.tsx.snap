// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link /> should render successfully 1`] = `
.c0 {
  display: block;
  color: white;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-bottom: 1px solid white;
}

<div>
  <a
    class="c0"
    href="https://dummy-link-url"
  >
    link-text
  </a>
</div>
`;

exports[`<Link /> should render successfully as a button 1`] = `
.c0 {
  display: block;
  color: white;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-bottom: 1px solid white;
  background-color: #3e8efa;
  text-transform: uppercase;
  padding: 12px 30px;
  border: none;
  border-radius: 3px;
}

<div>
  <a
    class="c0"
    href="https://dummy-link-url"
  >
    link-text
  </a>
</div>
`;
